src/components/account/login.vue

```diff
@@ -177,6 +186,11 @@ export default {
           (code === 401.2 ? this.$t('problem.401_2') : null)
       })
         .then(() => logIn(this.container, true))
+        .then(() => {
+          if (this.showMailingListOptIn) {
+            this.currentUser.preferences.site.mailingListOptIn = this.mailingListOptIn;
+          }
+        })
         .then(() => {
```

Comment:
I don't think this needs to be its own `.then()` separate from the next `.then()`. The new `.then()` callback doesn't return a promise, so the next `.then()` doesn't have anything to wait on.